source/Calamari.Shared/Integration/FileSystem/FreeSpaceChecker.cs

```diff
@@ -0,0 +1,61 @@
+using System;
+using System.IO;
+using Calamari.Commands.Support;
+using Calamari.Deployment;
+using Calamari.Integration.Processes;
+using Octostache;
+
+namespace Calamari.Integration.FileSystem
+{
+    public interface IFreeSpaceChecker
+    {
+        void EnsureDiskHasEnoughFreeSpace(string directoryPath);
+    }
+
+    public class FreeSpaceChecker : IFreeSpaceChecker
+    {
+        readonly ICalamariFileSystem fileSystem;
+        readonly CalamariVariableDictionary variables;
+
+        public FreeSpaceChecker(ICalamariFileSystem fileSystem, CalamariVariableDictionary variables)
```

Comment:
Something feels off about using this bag of variables. Could we initialise some kind of settings class that gets initialised from the variables at startup? The intention being that less places know about and read from variables?

Reply:
Yes, that is the intention as the refactor progresses. The Variables are actually already in the container as well.